vue-app/src/store/index.ts

```diff
@@ -127,11 +127,9 @@ export const mutations = {
       } else {
         state.cart.push(addedItem)
       }
-    } else if (state.cart[itemIndex].isCleared) {
-      // Restore cleared item
-      Vue.set(state.cart, itemIndex, addedItem)
     } else {
-      throw new Error('item is already in the cart')
+      console.warn('item is already in the cart') /* eslint-disable-line no-console */
+      Vue.set(state.cart, itemIndex, addedItem)
```

Comment:
@samajammin here is the overwrite of the item. As you can see, now we are doing the overwrite to all the items, not just the cleared ones.